python/GafferImageUI/CatalogueUI.py

```diff
@@ -371,28 +375,55 @@ def __addClicked( self, *unused ) :
 
 	def __removeClicked( self, *unused ) :
 
-		index = self.__indexFromSelection()
+		indices = self.__indicesFromSelection()
 
 		# If the user repeatedly clicks the delete button, we might end up in a
 		# state, where selection hasn't been restored yet. In that case we
 		# can't delete anything and will ignore the request.
-		if index is None :
+		if not indices :
 			return
 
 		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
-			self.__images().removeChild( self.__images()[index] )
-			self.getPlug().setValue( max( 0, index - 1 ) )
+			for i, index in enumerate( reversed( sorted( indices ) ) ) :
```

Comment:
`i` is unused - remove the `enumerate()`

Reply:
I believe this is removed in the next commit already. Apologies for the confusion - should've squashed that hunk into the previous commit.